ethereum/contracts/BeefyClient.sol

```diff
@@ -0,0 +1,474 @@
+     * @param version version of the leaf type
+     * @param parentNumber parent number of the block this leaf describes
+     * @param parentHash parent hash of the block this leaf describes
+     * @param parachainHeadsRoot merkle root of all parachain headers in this block
+     * @param nextAuthoritySetId validator set id that will be part of consensus for the next block
+     * @param nextAuthoritySetLen length of that validator set
+     * @param nextAuthoritySetRoot merkle root of all public keys in that validator set
+     */
+    struct MMRLeaf {
+        uint8 version;
+        uint32 parentNumber;
+        bytes32 parentHash;
+        uint64 nextAuthoritySetId;
+        uint32 nextAuthoritySetLen;
+        bytes32 nextAuthoritySetRoot;
+        bytes32 parachainHeadsRoot;
+    }
+
+    /* State */
+
+    bytes32 public latestMMRRoot;
+    uint64 public latestBeefyBlock;
+
+    struct ValidatorSet {
+        uint256 id;
+        bytes32 root;
+        uint256 length;
+    }
+
+    ValidatorSet public currentValidatorSet;
+    ValidatorSet public nextValidatorSet;
+
+    uint256 public nextID;
+    mapping(uint256 => ValidationData) public validationData;
+
+    /* Constants */
+
+    // Used for calculating minimum number of required signatures
+    uint256 public constant THRESHOLD_NUMERATOR = 3;
+    uint256 public constant THRESHOLD_DENOMINATOR = 250;
+
+    uint64 public constant BLOCK_WAIT_PERIOD = 3;
+
+    /**
+     * @notice Deploys the BeefyClient contract
+     */
+    constructor() {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        nextID = 0;
+    }
+
+    // Once-off post-construction call to set initial configuration.
+    function initialize(
+        uint64 _startingBeefyBlock,
+        ValidatorSet calldata _initialValidatorSet,
+        ValidatorSet calldata _nextValidatorSet
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        latestBeefyBlock = _startingBeefyBlock;
+        currentValidatorSet = _initialValidatorSet;
+        nextValidatorSet = _nextValidatorSet;
+
+        // drop admin privileges
+        renounceRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /* Public Functions */
+
+    /**
+     * @notice Executed by the incoming channel in order to verify commitment
+     * @param leafHash contains the merkle leaf to be verified
+     * @param proof contains simplified mmr proof
+     */
+    function verifyMMRLeafProof(bytes32 leafHash, MMRProof calldata proof)
+        external
+        view
+        returns (bool)
+    {
+        return MMRProofVerification.verifyLeafProof(latestMMRRoot, leafHash, proof);
+    }
+
+    /**
+     * @notice Executed by the prover in order to begin the process of block
+     * acceptance by the light client
+     * @param commitmentHash contains the commitmentHash signed by the validator(s)
+     * @param validatorClaimsBitfield a bitfield containing a membership status of each
```

Comment:
new param validatorSetID needs comment

Reply:
Updated all docstrings to be sync with latest code.